src/library-authoring/generic/filter-by-published/index.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { useLibraryContext } from '../../common/context/LibraryContext';
+import { FilterByPublished, PublishStatus } from '../../../search-manager';
+
+const LibraryFilterByPublished : React.FC<Record<never, never>> = () => {
```

Comment:
Nit: it would be good to have a JSDoc comment here explaining why we have this component ("When browsing library content for insertion into a course, we only show published content. In that case, there is no need for a 'Never Published' filter, which will never show results. This component removes that option from FilterByPublished when not relevant.")

Reply:
Updated: https://github.com/openedx/frontend-app-authoring/pull/1947/commits/df819d36c76c7d05059efb4538b470729f96cd59